gpiocontrol/source/objc/GPIOControlDemo/GPIOTableViewController.m

```diff
@@ -68,29 +66,24 @@ - (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(N
 - (IBAction)statusSwitchChanged:(UISwitch *)sender {
     ConnectivityChecker *checker = [[ConnectivityChecker alloc] init];
     if ([checker isConnected]) {
-        GpioStatus *status = [self.gpioStatusArray objectAtIndex:sender.tag];
-        status.status = sender.on;
-        id <KaaClient> client = [KaaProvider getClient];
-        EventFamilyFactory *eventFamilyFactory = [client getEventFamilyFactory];
-        RemoteControlECF *ecf = [eventFamilyFactory getRemoteControlECF];
-        GpioToggleRequest *request = [[GpioToggleRequest alloc] init];
-        request.gpio = self.gpioStatusArray[sender.tag];
-        [ecf sendGpioToggleRequest:request to:self.device.kaaEndpointId];
+        [self sendToggleRequest:sender];
     } else {
-        UIAlertController *alertController = [UIAlertController
-                                              alertControllerWithTitle:@"Connection status"
-                                              message:@"No connection"
-                                              preferredStyle:UIAlertControllerStyleAlert];
-        UIAlertAction* cancel = [UIAlertAction actionWithTitle:@"Cancel" style:UIAlertActionStyleDefault
-                                                       handler:^(UIAlertAction * action) {
-                                                           [alertController dismissViewControllerAnimated:YES completion:nil];
-                                                       }];
-        [alertController addAction:cancel];
-        sender.on = !sender.on;
-
-        [self presentViewController:alertController animated:YES completion:nil];
+        [self presentViewController:[ConnectionAlert noConnectionAlert]
+                           animated:YES
+                         completion:nil];
     }
 }
 
+- (void)sendToggleRequest:(UISwitch *)sender {
+    RemoteControlECFGpioStatus *status = [self.gpioStatusArray objectAtIndex:sender.tag];
+    status.status = sender.on;
+    id <KaaClient> client = [KaaClientManager sharedManager].kaaClient;
```

Comment:
Please don't put a space between an object type and the protocol it conforms to(in order to code guide